src/context.rs

```diff
@@ -979,6 +988,11 @@ impl<'context> WidgetContext<'context> {
         if let Some(mode) = overridden_theme_mode {
             context.cache.theme_mode = mode.get_tracking_redraw(&context);
         }
+        if let Some(locale) = overridden_locale {
+            context.locale = locale
+        }
+        context.locale.invalidate_when_changed(&context);
```

Comment:
We shouldn't be invalidating every context. This would force every widget context to invalidate even if the locale isn't used by a given widget.